testtools/matchers/_impl.py

```diff
@@ -53,7 +62,49 @@ def __str__(self):
         raise NotImplementedError(self.__str__)
 
 
-class Mismatch(object):
+Matcher = IMatcher
+
+
+class IMismatch(object):
+    """A mismatch detected by an IMatcher.
+
+    Describes the protocol that mismatches are required to implement.
+    """
+
+    # XXX: jml would like to extend this interface to include the thing that
```

Comment:
@jml Just for extra clarity, does "include the thing that" mean "the matchee argument passed to match()"?

From a very quick look at the code it seems that implementations of IMatcher typically pass or could pass the matchee object to the constructor of an IMismatch, in order to have describe() take it into account.

Why do we need to pass it to describe()?

An example of the "composition" use case you allude to would be appreciated.